src/components/vue-form-renderer.vue

```diff
@@ -205,42 +207,45 @@ export default {
         this.$emit('submit', this.transientData);
       }
     },
-    validateElements(elements) {
-      elements.forEach(element => {
-        if (element.validator && element.validator.errorCount !== 0) {
-          this.valid = false;
-          this.errors.push(element.validator.errors.errors);
+    getDataAndRules(items) {
+      let data = {}, rules = {};
+      items.forEach(item => {
+
+        if (Array.isArray(item)) {
+          const [data1, rules1] = this.getDataAndRules(item);
+          data = {...data, ...data1};
+          rules = {...rules, ...rules1};
         }
-      });
-    },
-    validateContainer(container) {
-      if (container.$refs && container.$refs.container) {
-        this.validateContainer(container.$refs.container);
-      }
-      container.forEach(element => {
-        if (element.$refs && element.$refs.elements) {
-          this.validateElements(element.$refs.elements);
+
+        if (item.items) {
+          const [data2, rules2] = this.getDataAndRules(item.items);
+          data = {...data, ...data2};
+          rules = {...rules, ...rules2};
+        }
+
+        if (item.config && item.config.name && item.config.validation) {
+          data[`${item.config.name}`] = this.data[item.config.name];
+          rules[`${item.config.name}`] = item.config.validation;
```

Comment:
It looks like this loop is doing three different things. Two of them look the same (the `Array.isArray(item)` and `item.items` `if` blocks seem to do the same thing); so there's possibly some duplication that can be eliminated.

See if you can split it up into three different `forEach` loops to improve readability. Once you have it broken down, you may also find there's a nice way to refactor it back into a single method.

Think about readability and testability. This  method may also be moved to a validator factory, as I suggest in my comment below.